File: android/Detonator/src/main/java/com/iconshot/detonator/tree/Target.java
package com.iconshot.detonator.tree;

import android.view.View;
import android.view.ViewParent;

import com.iconshot.detonator.layout.ViewLayout;

public class Target {
    private final ViewLayout view;
    private int index;

    public Target(ViewLayout view, int index) {
        this.view = view;
        this.index = index;
    }

    public void insert(View child) {
        ViewParent parent = child.getParent();

        View currentChild = view.getChildAt(index);

        if (currentChild != null) {
            if (child != currentChild) {
                if (parent != null) {
                    remove(child);
                }

                view.addView(child, index);
            }
        } else {
            if (parent != null) {
                remove(child);
            }

            view.addView(child);
        }

        index++;
    }

    public void remove(View child) {
        view.removeView(child);
    }
}

File: ios/Detonator/Sources/Detonator/Tree/Target.swift
import UIKit

class Target {
    public final let view: UIView
    private var index: Int
    
    init(view: UIView, index: Int) {
        self.view = view
        self.index = index
    }
    
    public func insert(child: UIView) {
        let currentChild = index < view.subviews.count
            ? view.subviews[index]
            : nil
        
        if currentChild != nil {
            if child != currentChild {
                view.insertSubview(child, at: index)
            }
        } else {
            view.addSubview(child)
        }
        
        index += 1
    }
    
    public func remove(child: UIView) {
        child.removeFromSuperview()
    }
}
